mozilla-ai/document-to-podcast
# Fix codespace issue

# What's changing

Currently the `.devcontainer.json` is ignored because its under the `.github` directory. It needs to be either in `root` or `.devcontainer/` according to the [docs](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/introduction-to-dev-containers#default-configuration-selection-during-codespace-creation).

This meant that the default configuration of codespaces for a machine with 16GB RAM was ignored, and instead was using the 8GB RAM machine which is not sufficient to properly run the demo app in codespaces. Thanks @ameckes1229 for the discovery!

Also the pre-installation step used in the `.devcontainer.json` was deprecated and now its updated to use the `.github/setup.sh` script.

# How to test it

Steps to test the changes:

1. Create default codespace (just by clicking `+`)
2. Click `Code` in the repo -> `...` in the active codespace -> `Change machine type` ->  Verify that the machine selected is the 16GB one
3. Run the demo in codespaces **without first installing anything**. The installation of the package should already been taken care of automatically when setting up the codespace.

# Additional notes for reviewers

~~Not sure how we can test this before merging to main :/~~

Edit: The only way to test this (to my understanding) is by using [pre-build codespaces](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds), but we need to discuss how or if we can set this up on org level. About testing codespaces configuration changes [here](https://docs.github.com/en/codespaces/prebuilding-your-codespaces/testing-dev-container-changes)

# I already...
